src/libsync/discovery.cpp

```diff
@@ -531,6 +565,21 @@ void ProcessDirectoryJob::processFileAnalyzeRemoteInfo(
             if (isVfsWithSuffix())
                 addVirtualFileSuffix(tmp_path._original);
         }
+        // a virtual file is on disk but is not in the database
+        const bool fileOnDiskIsVirtual = localEntry.isValid() && localEntry.isVirtualFile;
+        // a new virtual file
+        const bool isNewVirtualFileNotOnDiskYet = !localEntry.isValid() && item->_type == ItemTypeVirtualFile;
+
+        if ((fileOnDiskIsVirtual || isNewVirtualFileNotOnDiskYet)
```

Comment:
why is this condition not just `opts._vfs->mode() != Vfs::Off && !item->_encryptedFileName.isEmpty()` ?
I fail to understand why in some cases we would remove the tag size and in other cases we would keep it